app/src/main/java/com/pavesid/androidacademy/services/MoviesWorker.kt

```diff
@@ -0,0 +1,25 @@
+package com.pavesid.androidacademy.services
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.pavesid.androidacademy.repositories.MoviesRepository
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class MoviesWorker(context: Context, workerParams: WorkerParameters) :
+    CoroutineWorker(context, workerParams) {
+
+    @Inject
+    lateinit var repository: MoviesRepository
```

Comment:
That doesn't work. 
**UninitializedPropertyAccessException: lateinit property repository has not been initialized**

Reply:
Done.